clang/lib/Sema/SemaExpr.cpp

```diff
@@ -12901,24 +12901,24 @@ QualType Sema::GetSignedVectorType(QualType V) {
     return Context.getExtVectorType(Context.LongLongTy, VTy->getNumElements());
   }
 
-  if (TypeSize == Context.getTypeSize(Context.Int128Ty))
-    return Context.getVectorType(Context.Int128Ty, VTy->getNumElements(),
-                                 VectorKind::Generic);
-  if (TypeSize == Context.getTypeSize(Context.LongLongTy))
-    return Context.getVectorType(Context.LongLongTy, VTy->getNumElements(),
-                                 VectorKind::Generic);
-  if (TypeSize == Context.getTypeSize(Context.LongTy))
-    return Context.getVectorType(Context.LongTy, VTy->getNumElements(),
-                                 VectorKind::Generic);
   if (TypeSize == Context.getTypeSize(Context.IntTy))
     return Context.getVectorType(Context.IntTy, VTy->getNumElements(),
                                  VectorKind::Generic);
+  if (TypeSize == Context.getTypeSize(Context.SignedCharTy))
```

Comment:
Why does the ordering of these matter since they are doing direct comparison of size?  The only thing that seems to matter here is the order of long vs long-long, since they can be the same size, right?  So only those two need to swap?

But otherwise putting these in any sort of order should have some sort of semblence of ordering if we're going for that.  For example, int before char is... awkward.  We should probably make sure we're doing this in the 'rank order' from C/C++.

Reply:
I also think making `int` before `char` is awkward, but if we want to be totally consistent with GCC, following their rule is possibly the best way...

Reply:
Outdated.